enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+---
+title: multiple-networks-vrfs
+authors:
+  - "@trozet"
+  - "@qinqon"
+reviewers:
+  - "@tsurya"
+  - "@danwinship"
+  - "@fedepaol"
+  - "@maiqueb"
+  - "@jcaamano"
+  - "@Miciah"
+  - "@dceara"
+  - "@dougbtv"
+approvers:
+  - "@tsurya"
+  - "@jcaamano"
+api-approvers:
+  - "None"
+creation-date: 2024-05-03
+last-updated: 2024-05-28
+tracking-link: https://issues.redhat.com/browse/SDN-4789
+---
+
+# Multiple Networks/VRFs - User Defined Native Network Isolation
+
+## Summary
+
+OVN-Kubernetes today allows multiple different types of networks per secondary network: layer 2, layer 3, or localnet.
+Pods can be connected to different networks without discretion. For the primary network, OVN-Kubernetes only supports all
+pods connecting to the same layer 3 virtual topology. The scope of this effort is to bring the same flexibility of the
+secondary network to the primary network. Therefore, pods are able to connect to different types of networks that are
+independent of networks that other pods may connect to.
+
+## Terminology
+
+* **Primary Network** - The network which is used as the default gateway for the pod. Typically recognized as the eth0
+interface in the pod.
+* **Secondary Network** - An additional network and interface presented to the pod. Typically created as an additional
+Network Attachment Definition (NAD), leveraging Multus.
+* **Cluster Default Network** - This is the routed OVN network that pods attach to by default today as their primary network.
+The pods default route, service access, as well as kubelet probe are all served by the interface (typically eth0) on this network.
+* **User-Defined Network** - A network that may be primary or secondary, but is declared by the user.
+
+## Motivation
+
+As users migrate from OpenStack to Kubernetes, there is a need to provide network parity for those users. In OpenStack,
+each tenant (akin to Kubernetes namespace) by default has a layer 2 network, which is isolated from any other tenant.
+Connectivity to other networks must be specified explicitly as network configuration via a Neutron router. In Kubernetes
+the paradigm is opposite, by default all pods can reach other pods, and security is provided by implementing Network Policy.
+Network Policy can be cumbersome to configure and manage for a large cluster. It also can be limiting as it only matches
+TCP, UDP, and SCTP traffic. Furthermore, large amounts of network policy can cause performance issues in CNIs. With all 
```

Comment:
good point! yes, since the pod will be wired to a topology that is separate from the others, it will be guaranteed to have network isolation on CNI ADD.